icu/icu4c/source/common/uprefs.h

```diff
@@ -0,0 +1,29 @@
+// © 2016 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#ifndef UPREFS_H
+#define UPREFS_H
+
+#include "unicode/platform.h"
+#include "unicode/utypes.h"
+#if U_PLATFORM_USES_ONLY_WIN32_API
```

Comment:
I think we should have the code in uprefs.h and uprefs.cpp conditional on `UCONFIG_USE_WINDOWS_PREFERENCES_LIBRARY` being enabled.

So this would become:
```suggestion
#if U_PLATFORM_USES_ONLY_WIN32_API && UCONFIG_USE_WINDOWS_PREFERENCES_LIBRARY
```

(And also the `#endif` below too)